In [ ]:
import pandas as pd

#pick only the strains used in the paper
import pandas as pd

all_strains_path = "../literature/all_strains_form_supplement.csv"
chosen_strains_path = "../literature/chosen_strains_from_supplement.csv"

df = pd.read_csv(all_strains_path, header=None, names=["Species", "Strain"])

# pick only lines ending with T
df_chosen = df[df['Strain'].str.strip().str.endswith('T')].copy()

# remove T from the end
df_chosen['Strain'] = df_chosen['Strain'].str.strip().str.rstrip('T')

df_chosen.to_csv(chosen_strains_path, index=False, header=False)

In [22]:
import os
import pandas as pd

accessions_path = "accessions.csv"
data_dir = "../data"

# load accessions
df = pd.read_csv(accessions_path)

# check files in folder
files_in_data = os.listdir(data_dir)

# identify missing
missing = []
for _, row in df.iterrows():
    # reconstruct expected filename
    expected_name = row['Species'].replace(" ", "_").replace(".", "").replace(":", "") + ".faa"
    if expected_name not in files_in_data:
        missing.append(expected_name)

print(f"Missing files: {missing}")

Missing files: ['Bifidobacterium_saeculare.faa']


Bifidobacterium saeculare is missing because it has been reclassified as a subspecies of Bifidobacterium gallinarum and was unlisted from the NCBI database.

Scardovia inopinata was addded manually because it's not present in the supplementary table from the paper.


In [23]:
df = pd.read_csv("accessions.csv")
print(df[df['Strain'] != df['Found_Strain_NCBI']])

                                             Species     Strain  \
50                        Bifidobacterium asteroides  LMG 10735   
51                           Bifidobacterium dentium  LMG 11045   
52                           Bifidobacterium indicum  LMG 11587   
53                         Bifidobacterium saeculare  LMG 14934   
54  Bifidobacterium thermacidophilum subsp. porcinum  LMG 21689   
55                               Scardovia inopinata  JCM 12537   

   Found_Strain_NCBI        Accession  
50         DSM 20089  GCF_002715865.1  
51         DSM 20436  GCF_001025055.1  
52         DSM 20214  GCF_000706765.1  
53          DSM 6531  GCF_003951865.1  
54         DSM 17755  GCF_000771045.1  
55         DSM 10107  GCF_000154065.1  


Some species are listed under different names in the paper and NCBI, but sources listed below support that those are just the same species under different names.

| Species Name | Strain (Paper) | Strain (NCBI) | Equivalent According To |
| :--- | :--- | :--- | :--- |
| *Bifidobacterium asteroides* | LMG 10735 | DSM 20089 | [BacDive ID 1688](https://bacdive.dsmz.de/strain/1688) |
| *Bifidobacterium dentium* | LMG 11045 | DSM 20436 | [BCCM/LMG Catalogue](https://bccm.belspo.be/page/lmg-catalogue-display/fields/name/LMG%2011045) |
| *Bifidobacterium indicum* | LMG 11587 | DSM 20214 | [KEGG Organism Database](https://www.kegg.jp/kegg-bin/show_organism?org=bii) |
| *Bifidobacterium thermacidophilum* subsp. *porcinum* | LMG 21689 | DSM 17755 | [JCM Catalogue (RIKEN)](https://www.jcm.riken.jp/cgi-bin/jcm/jcm_keyword?AN=Bifidobacterium&BN=thermacidophilum&CN=porcinum&DN=) |
| *Scardovia inopinata* | JCM 12537 | DSM 10107 | [SeqCode Registry](https://registry.seqco.de/names/18742) |